### PR TITLE
upgrade to javafx 18

### PR DESCRIPTION
**Pull request by florendg**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Setup Java
         uses: actions/setup-java@v1
         with:
-          java-version: 16
+          java-version: 17
```
Comment (MairwunNx) [inline .github/workflows/build.yml]: I think you can use 18 java version number here.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https://services.gradle.org/distributions/gradle-7.2-bin.zip
+distributionUrl=https://services.gradle.org/distributions/gradle-7.4.1-bin.zip
```
Comment (MairwunNx) [inline gradle/wrapper/gradle-wrapper.properties]: I think you can use newest version of gradle wrapper (7.4.2)

File: app/build.gradle.kts [hunk 3]
```diff
@@ -45,7 +45,8 @@ application {
 java {
     modularity.inferModulePath.set(true)
     toolchain {
-        languageVersion.set(JavaLanguageVersion.of(17))
+        vendor.set(JvmVendorSpec.ORACLE)
```
Comment (MairwunNx) [inline app/build.gradle.kts]: Probably at now Oracle distributive can be not find by gradle, we can use AZUL, it's works fine on GitHub CI

Comment (florendg) [reply app/build.gradle.kts]: Removed the requirement for a specific vendor

**Pull request merged**